addons/account/wizard/accrued_orders.py

```diff
@@ -0,0 +1,138 @@
+    journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        default=_default_journal_id,
+        domain=[('type', '=', 'general')],
+        readonly=False,
+        required=True,
+        string='Journal',
+    )
+    date = fields.Date(default=fields.Date.today(), required=True)
+    reversal_date = fields.Date(default=fields.Date.today() + relativedelta(days=1), required=True)
+    account_id = fields.Many2one(
+        'account.account',
+        required=True,
+        string='Accrual Account',
+        domain=_get_account_domain,
+    )
+    preview_data = fields.Text(compute='_compute_preview_data')
+    allowed_account_ids = fields.Many2many('account.account', compute='_compute_allowed_account_ids')
+
+    @api.depends('date', 'journal_id', 'account_id')
+    def _compute_preview_data(self):
+        for record in self:
+            preview_vals = [self.env['account.move']._move_dict_to_preview_vals(self._compute_move_vals(), record.company_id.currency_id)]
+            preview_columns = [
+                {'field': 'account_id', 'label': _('Account')},
+                {'field': 'name', 'label': _('Label')},
+                {'field': 'debit', 'label': _('Debit'), 'class': 'text-right text-nowrap'},
+                {'field': 'credit', 'label': _('Credit'), 'class': 'text-right text-nowrap'},
+            ]
+            record.preview_data = json.dumps({
+                'groups_vals': preview_vals,
+                'options': {
+                    'columns': preview_columns,
+                },
+            })
+
+    def _compute_move_vals(self):
+        move_lines = []
+        orders = []
+        if self.env.context.get('active_model') == 'purchase.order':
+            orders = self.env['purchase.order'].browse(self.env.context.get('active_ids'))
+            is_purchase = True
+        else:
+            orders = self.env['sale.order'].browse(self.env.context.get('active_ids'))
+            is_purchase = False
+
+        if len(orders.mapped('company_id')) > 1:
+            raise UserError(_('Entries can only be created for one company at a time.'))
+
+        for order in orders:
+            amt = 0.0
+            precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+            for order_line in order.order_line:
+                if not float_is_zero(order_line.qty_to_invoice, precision_digits=precision):
+                    amt += order_line.qty_to_invoice * order_line.price_unit
+            if not float_is_zero(amt, precision_digits=self.company_id.currency_id.decimal_places):
+                move_lines.append(Command.create({
+                    'name': _('Accrued for %s', order.name),
+                    'debit': amt if is_purchase else 0.0,
+                    'credit': 0.0 if is_purchase else amt,
+                    'account_id': (
+                        order.partner_id.property_account_receivable_id.id if
```

Comment:
in case in multi company dont forget to pass :
`order.partner_id.with_company(order.company_id).property_account_receivable_id.id`